Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -232,13 +232,16 @@ final class CacheController: CacheControlling {
         for graph: Graph,
         cacheProfile: TuistGraph.Cache.Profile,
         cacheOutputType: CacheOutputType,
-        includedTargets: [String],
+        includedTargets: Set<String>,
         dependenciesOnly: Bool
     ) throws -> [(GraphTarget, String)] {
+        // When `dependenciesOnly` is true, there is no need to compute `includedTargets` hashes
+        let excludedTargets = dependenciesOnly ? includedTargets : []
         let hashesByCacheableTarget = try cacheGraphContentHasher.contentHashes(
             for: graph,
             cacheProfile: cacheProfile,
-            cacheOutputType: cacheOutputType
+            cacheOutputType: cacheOutputType,
+            excludedTargets: excludedTargets
```

Comment:
I would say we don't need anymore the filter at line 261. Could you double-check that?
Since we will now skip the hashing for those targets they should not be available anymore in the `hashesByCacheableTarget`.

Reply:
yep, I have removed it 🚀